sklearn/manifold/mds.py

```diff
@@ -108,14 +110,21 @@ def _smacof_single(dissimilarities, metric=True, n_components=2, init=None,
                                    (disparities ** 2).sum())
 
         # Compute stress
-        stress = ((dis.ravel() - disparities.ravel()) ** 2).sum() / 2
+        if masked:
+            stress = ((dis - disparities).compressed() ** 2).sum() / 2
+        else:
+            stress = ((dis.ravel() - disparities.ravel()) ** 2).sum() / 2
```

Comment:
This isn't necessary, `MaskedArray.sum()` already does the right thing here

Reply:
I think there was a case where one of the two arrays is not masked and I was getting arrays of different lengths that couldn't be differenced.

Reply:
That doesn't make any sense to me. `ravel` returns the same length array irrespective of the mask.

It's also unecessary, and `stress = ((dis - disparities) ** 2).sum() / 2` would work fine here